tools/ci/macos_display_configuration.py

```diff
@@ -0,0 +1,237 @@
+import argparse
+import sys
+from typing import Any, NewType, Optional, Tuple
+
+from Cocoa import NSURL
+from ColorSync import (
+    CGDisplayCreateUUIDFromDisplayID,
+    ColorSyncDeviceSetCustomProfiles,
+    kColorSyncDeviceDefaultProfileID,
+    kColorSyncDisplayDeviceClass,
+)
+from Quartz import (
+    CGBeginDisplayConfiguration,
+    CGCancelDisplayConfiguration,
+    CGCompleteDisplayConfiguration,
+    CGConfigureDisplayWithDisplayMode,
+    CGDisplayCopyAllDisplayModes,
+    CGDisplayCopyDisplayMode,
+    CGDisplayModeGetHeight,
+    CGDisplayModeGetIOFlags,
+    CGDisplayModeGetPixelHeight,
+    CGDisplayModeGetPixelWidth,
+    CGDisplayModeGetRefreshRate,
+    CGDisplayModeGetWidth,
+    CGDisplayModeIsUsableForDesktopGUI,
+    CGDisplayModeRef,
+    CGGetOnlineDisplayList,
+    kCGConfigurePermanently,
+    kCGErrorSuccess,
+)
+
+# Display mode flags
+kDisplayModeDefaultFlag = 0x00000004  # noqa: N816
+
+# Create a new type for display IDs
+CGDirectDisplayID = NewType("CGDirectDisplayID", int)
+
+
+def get_pixel_size(mode: CGDisplayModeRef) -> Tuple[int, int]:
+    return (CGDisplayModeGetPixelWidth(mode), CGDisplayModeGetPixelHeight(mode))
+
+
+def get_size(mode: CGDisplayModeRef) -> Tuple[int, int]:
+    return (CGDisplayModeGetWidth(mode), CGDisplayModeGetHeight(mode))
+
+
+def calculate_mode_similarity_score(
+    mode: CGDisplayModeRef, current_mode: CGDisplayModeRef
+) -> int:
+    current_size = get_size(current_mode)
+    current_pixel_size = get_pixel_size(current_mode)
+    current_refresh_rate = CGDisplayModeGetRefreshRate(current_mode)
+    current_flags = CGDisplayModeGetIOFlags(current_mode)
+
+    size = get_size(mode)
+    pixel_size = get_pixel_size(mode)
+    refresh_rate = CGDisplayModeGetRefreshRate(mode)
+    flags = CGDisplayModeGetIOFlags(mode)
+
+    differences = 0
+
+    if size != current_size:
+        differences += 1
+    if pixel_size != current_pixel_size:
+        differences += 1
+    if refresh_rate != current_refresh_rate:
+        differences += 1
+
+    # Count how many individual flags are changing (XOR then count bits)
+    changed_flags = flags ^ current_flags
+    if sys.version_info >= (3, 10):
+        differences += changed_flags.bit_count()
+    else:
+        differences += bin(changed_flags).count("1")
+
+    return differences
+
+
+def find_best_unscaled_mode(display_id: CGDirectDisplayID) -> CGDisplayModeRef:
+    current_mode: Optional[CGDisplayModeRef] = CGDisplayCopyDisplayMode(display_id)
```

Comment:
Isn't the python default arguments evaluated at the function definition evaluation time?
Wouldn't this would mean that if the script would change the mode, the current_mode would still refer to the mode that was current when the script top-level was evaluated? Given the function is written in reusable, generic manner, this is highly unintuitive.